test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Utility/FileWatchers/SolutionConfigFileWatcherTests.cs

```diff
@@ -0,0 +1,170 @@
+            obtained.Should().BeTrue();
+        }
+
+        [Theory]
+        [InlineData("nuget.config", "nuget.config.old")]
+        [InlineData("nuget.config.old", "nuget.config")]
+        public void FileChanged_WhenFileRenamed_NotificationReceived(string filename1, string filename2)
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            string configPath1 = Path.Combine(testDirectory.Path, filename1);
+            string configPath2 = Path.Combine(testDirectory.Path, filename2);
+            File.WriteAllText(configPath1, string.Empty);
+            ManualResetEventSlim mre = new(initialState: false, spinCount: 0);
+            using SolutionConfigFileWatcher target = new(testDirectory.Path);
+            target.FileChanged += (s, e) => mre.Set();
+
+            // Act
+            mre.Wait(0).Should().BeFalse();
+            File.Move(configPath1, configPath2);
+            // filesystem watchers are not real time, so we need to give a litle time for all the async file IO to happen
+            bool obtained = mre.Wait(TimeSpan.FromSeconds(10));
+
+            // Assert
+            obtained.Should().BeTrue();
+        }
+
+        [Fact]
+        public void FileChanged_WhenFileCreatedInParentDirectory_NotificationReceived()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            var solutionDirectory = Path.Combine(testDirectory, "solution");
+            using SolutionConfigFileWatcher target = new(solutionDirectory);
+            ManualResetEventSlim mre = new(initialState: false, spinCount: 0);
+            target.FileChanged += (s, e) => mre.Set();
+            string configPath = Path.Combine(testDirectory.Path, Settings.DefaultSettingsFileName);
+
+            // Act
+            File.WriteAllText(configPath, string.Empty);
+            // filesystem watchers are not real time, so we need to give a litle time for all the async file IO to happen
+            bool obtained = mre.Wait(TimeSpan.FromSeconds(10));
+
+            // Assert
+            obtained.Should().BeTrue();
+        }
+
+        [Fact]
+        public void FileChanged_NonConfigFileChanged_NotificationNotReceived()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            ManualResetEventSlim mre = new(initialState: false, spinCount: 0);
+            using (SolutionConfigFileWatcher target = new(testDirectory.Path))
+            {
+                target.FileChanged += (s, e) => mre.Set();
+            }
+            string slnPath = Path.Combine(testDirectory.Path, "solution.sln");
+            string packagesConfigPath = Path.Combine(testDirectory.Path, "packages.config");
+            string nugetConfigBakPath = Path.Combine(testDirectory.Path, Settings.DefaultSettingsFileName + ".bak");
+
+            // Act
+            File.WriteAllText(slnPath, string.Empty);
```

Comment:
I had a hard time mentally parsing why this one is using `string.Empty`. I checked `WriteAllText` and it doesn't explain what an empty content argument actually does. Is it setting metadata like modified date and that's how this test works?
If it's that subtle, can you just comment that you're just using the fact that empty content is considered a modification, or somehow explain this so it isn't something a reader has to figure out when they look at this test?

Reply:
I see now. Thx!